nautobot/dcim/views.py

```diff
@@ -199,15 +201,42 @@ def get_selected_objects_parents_name(self, selected_objects):
 #
 
 
-class LocationTypeListView(generic.ObjectListView):
-    queryset = LocationType.objects.all()
-    filterset = filters.LocationTypeFilterSet
-    filterset_form = forms.LocationTypeFilterForm
-    table = tables.LocationTypeTable
-
-
-class LocationTypeView(generic.ObjectView):
+class LocationTypeUIViewSet(
+    view_mixins.ObjectDetailViewMixin,
+    view_mixins.ObjectListViewMixin,
+    view_mixins.ObjectEditViewMixin,
+    view_mixins.ObjectDestroyViewMixin,
+    view_mixins.ObjectBulkDestroyViewMixin,
+    view_mixins.ObjectBulkCreateViewMixin,  # 3.0 TODO: remove this mixin as it's no longer used
+    view_mixins.ObjectChangeLogViewMixin,
+    view_mixins.ObjectNotesViewMixin,
+):
     queryset = LocationType.objects.all()
+    filterset_class = filters.LocationTypeFilterSet
+    filterset_form_class = forms.LocationTypeFilterForm
+    table_class = tables.LocationTypeTable
+    form_class = forms.LocationTypeForm
+    serializer_class = serializers.LocationSerializer
+
+    object_detail_content = ObjectDetailContent(
+        panels=(
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+                fields="__all__",
+            ),
+            ObjectsTablePanel(
+                weight=100,
+                table_key="children_table",
+                table_title="Child Location Type(s)",
+            ),
+            ObjectsTablePanel(
+                weight=200,
+                table_key="locations_table",
```

Comment:
Presumably `get_extra_context` no longer needs to paginate `children_table` and `locations_table` since that's now handled by ObjectsTablePanel itself?